xcdat/dataset.py

```diff
@@ -371,21 +373,34 @@ def decode_time(dataset: xr.Dataset) -> xr.Dataset:  # noqa: C901
             except KeyError:
                 bounds = None
 
-            if bounds is not None and not _is_time_decoded(bounds):
-                decoded_bounds = _decode_time_bounds(ref_date, bounds, calendar, units)
-                ds = ds.assign({bounds.name: decoded_bounds})
+            if bounds is not None:
+                # `.values` converts the underlying array to `np.ndarray` if it
+                # is not already an `np.ndarray` (e.g., multi-file Datasets
+                # which store data variables as Dask Arrays). Calling `.values`
+                # multiple times can result in multiple conversion operations,
+                # leading to decreased runtime performance. To avoid this
+                # situation, we store `bounds.values` as a variable and reuse it
+                # for decoding (if necessary).
+                bounds_vals = bounds.values
+
+                if not _is_decoded(bounds_vals):
+                    decoded_bounds = _decode_time_bounds(
+                        ref_date, bounds, bounds_vals, calendar, units
+                    )
+                    ds = ds.assign({bounds.name: decoded_bounds})
```

Comment:
@pochedls, this should improve this PR's decoding performance. 

Please give this a try in your performance testing.

Reply:
Basic test structure:
```
import xcdat as xc
import xarray as xr

fn = '/p/css03/cmip5_css02/data/cmip5/output1/INM/inmcm4/historical/mon/atmos/Amon/r1i1p1/tas/1/tas_Amon_inmcm4_historical_r1i1p1_185001-200512.nc'

%timeit ds = xc.open_dataset(fn); ds.close()
%timeit ds = xr.open_dataset(fn, use_cftime=True); ds.close()
```

xarray
> 923 ms ± 27 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)

xcdat (most recent iteration of this PR)
> 1.71 s ± 3.78 ms per loop (mean ± std. dev. of 7 runs, 1 loop each) 

xcdat (current main branch)
> 1.71 s ± 2.25 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)

xcdat (v0.3.3)
> 1.89 s ± 32 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)